agents/xy_stage/xy_ocs_agent.py

```diff
@@ -0,0 +1,249 @@
+        """
+
+        with self.lock.acquire_timeout(timeout=3, job='move_x_cm') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start x move because lock held by {self.lock.job}")
+                return False
+            self.xy_stage.move_x_cm( params['distance'], params['velocity'])
+        
+        time.sleep(1)
+        while True:
+            ## data acquisition updates the moving field if it is running
+            if not self.take_data:
+                with self.lock.acquire_timeout(timeout=3, job='move_x_cm') as acquired:
+                    if not acquired:
+                        self.log.warn(f"Could not check because lock held by {self.lock.job}")
+                        return False, "Could not acquire lock"
+                    self.is_moving = self.xy_stage.moving
+            
+            if not self.is_moving:
+                break
+        return True, "X Move Complete"
+
+    def move_y_cm(self, session, params):
+        """
+        params: dict: { 'distance': float, 'velocity':float < 1.2}
+        """
+
+        with self.lock.acquire_timeout(timeout=3, job='move_y_cm') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start y move because lock held by {self.lock.job}")
+                return False, "could not acquire lock"
+            self.xy_stage.move_y_cm( params['distance'], params['velocity'])
+        
+        time.sleep(1)
+        while True:
+            ## data acquisition updates the moving field if it is running
+            if not self.take_data:
+                with self.lock.acquire_timeout(timeout=3, job='move_y_cm') as acquired:
+                    if not acquired:
+                        self.log.warn(f"Could not check for move because lock held by {self.lock.job}")
+                        return False, "could not acquire lock"
+                    self.is_moving = self.xy_stage.moving
+            if not self.is_moving:
+                break
+        return True, "Y Move Complete"
+
+ 
+    def set_position(self, session, params):
+        """
+        params: dict: {'position': (float, float)}
+        """
+        with self.lock.acquire_timeout(timeout=3, job='set_position') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not set position because lock held by {self.lock.job}")
+                return False, "Could not acquire lock"
+                        
+            self.xy_stage.position = params['position']
+        return True, "Position Updated"
+
+    def start_acq(self, session, params=None):
+        """
+        params: dict: {`sampling_frequency': float, sampling rate in Hz}
+        """
+        if params is None:
+            params = {}
+
+        
+        f_sample = params.get('sampling_frequency', self.sampling_frequency)
+        pm = Pacemaker(f_sample, quantize=True)
+
+        if not self.initialized:
+            self.init_xy_stage_task(session)
```

Comment:
I don't think this is explicitly forbidden... I just have never thought of calling an operation from another operation by just passing in the session object. @mhasself is this allowed? It might mess up the "status" flow of the session, which I think has to be ordered: starting < running < stopping < done. Since the `init_xy_stage_task` doesn't set the I don't think this will break, but I'm pretty sure it's not recommended.

Reply:
I think a more official way to do this would be to call the `self.agent`'s  [start](https://github.com/simonsobs/ocs/blob/develop/ocs/ocs_agent.py#L542) and [wait](https://github.com/simonsobs/ocs/blob/ff0c23995c3682b1a66969647c46304963759afb/ocs/ocs_agent.py#L603)  functions with the operation name

Reply:
Something is messing up there but I'll just raise an error if things are initialized.